src/core/scene/a-scene.js

```diff
@@ -265,6 +273,17 @@ module.exports.AScene = registerElement('a-scene', {
     },
 
     /**
+     * Handle exiting VR through other means than `<ESC>` key. For example, GearVR back button
+     * on Oculus Browser.
+     */
+    onVRPresentChange: {
+      value: function () {
+        if (!this.is('vr-mode')) { return; }
```

Comment:
can we have `vr-mode` set if `vrdisplaypresentchange` fires upon entering VR? if an A-Frame site wants to enter VR mode using the WebVR API directly, currently devs have to explicitly call `$('a-scene').enterVR()`; simply calling `navigator.getVRDisplays().then(displays => displays[0].requestPresent([{source: $('canvas[data-aframe-canvas]')}])` won't work.

this is actually necessary for properly entering VR mode upon page navigation (without the `auto-enter-vr` component, which _forces_ entering VR mode, not responding to the user's already being in VR mode).

there are cases where sites would want to handle entering VR mode by using the WebVR API directly, without needing to special case for A-Frame scenes vs. three.js scenes vs. others. examples: the [WebVR Agent](https://github.com/webvrrocks/webvr-agent), a JS user script/bookmarklet, a WebExtension/Chrome browser extension, etc.

this will require a couple changes to the `enterVR`. specifically, instead of calling [`this.effect.requestPresent(…)`](https://github.com/ngokevin/aframe/blob/c01a63ba/src/core/scene/a-scene.js#L197), we'll want to just check if `if (checkHeadsetConnected || this.isMobile)` and call `enterVRSuccess()`.